app/src/main/kotlin/com/hippo/ehviewer/image/Image.kt

```diff
@@ -53,12 +57,18 @@ import splitties.init.appCtx
 class Image private constructor(image: CoilImage, private val src: ImageSource) {
     val size = image.size
     val rect = when (image) {
-        is BitmapImageWithRect -> image.rect
-        else -> image.run { IntRect(0, 0, width, height) }
+        is BitmapImageWithExtraInfo -> image.rect
+        else -> with(image) { IntRect(0, 0, width, height) }
     }
+    var hasQrCode by mutableStateOf(
+        when (image) {
+            is BitmapImageWithExtraInfo -> image.hasQrCode
+            else -> false
+        },
+    )
```

Comment:
```
pages[index].status.value = if (image.hasQrCode) Page.State.BLOCKED else Page.State.READY
```
in `notifyPageSucceed` instead of making `hasQrCode` a state might be better.